### PR TITLE
add yield in resolve context

### PR DESCRIPTION
**Pull request by vankop**
- If `resolveContext.yield` presented will return resolve results there `resolveContext.yield(obj)`.
- In this case resolve callback will not return any result.
- If no result yielded error with log will be returned as always.

### CODE REVIEW

File: lib/getPaths.js [hunk 1]
```diff
@@ -6,6 +6,7 @@
 "use strict";
 
 module.exports = function getPaths(path) {
+	if (path === "/") return { paths: ["/"], seqments: [""] };
```
Comment (vankop) [inline lib/getPaths.js:9]: have find a small bug where several same results were returned.
without this line `{ paths: ["/", "/"] }` returned

**Comment by vankop** [comment]
maybe in `yield` case we need to use `forEachAll` instead of `forEachBail`, not sure that this make sense since anyway this improves only edge case performance.

File: test/yield.js [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			should(Array.from(fileDependencies.values()).sort()).be.eql([
+				"/",
+				"/a",
+				"/a/foo",
+				"/a/foo-2",
+				"/a/foo-2/b",
+				"/a/foo/b"
+			]);
+			should(Array.from(missingDependencies.values()).sort()).be.eql([
+				"/a/foo-2/b",
```
Comment (sokra) [inline test/yield.js:62]: Interesting that this is in `missingDependencies` and in `fileDependencies`.
I guess it tries "as file" first and "as directory" second, resulting in it exists as file but is missing as directory...
Doesn't seem to be directly caused by this PR, so probably not relavant.

File: test/yield.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+			should(Array.from(fileDependencies.values()).sort()).be.eql([
+				"/",
+				"/a",
+				"/a/foo",
+				"/a/foo/a",
+				"/b",
+				"/b/foo",
+				"/b/foo/a"
+			]);
+			should(Array.from(missingDependencies.values()).sort()).be.eql([
```
Comment (sokra) [inline test/yield.js:103]:
```suggestion
			should(Array.from(missingDependencies).sort()).be.eql([
```

Technically that `.values()` is not necessary, but nevermind.

**Review by sokra — APPROVED** [review]

**Comment by sokra** [comment]
Thanks
